app/jobs/import_facilities_job.rb

```diff
@@ -12,6 +12,10 @@ def perform(facilities)
       import_facilities << import_facility
     end
 
-    Facility.import!(import_facilities, validate: true)
+    ActiveRecord::Base.transaction do
+      Facility.import!(import_facilities, validate: true)
+      # import! can't run callbacks, so we manually ensure that we create regions
+      import_facilities.each(&:make_region) if Flipper.enabled?(:regions_prep)
+    end
```

Comment:
Drats. Even though we used model callbacks, looks like there was another entryway huh 😅  I'm not suggesting that one way was better than the other, just sympathizing 😄

Reply:
It sucks. But regardless of the region creation approach, we'd have to do this here, so I'm ok pawning this off as an "edge-case" since we're using this separate gem to make these uploads faster.